### PR TITLE
 fix: set orient of bar in boxpolot macro

### LINKED ISSUE #4849: The boxes in Boxplot have incorrect size and offset when setting both fields quantitative.
```json
{
  "mark": {"type": "boxplot"},
  "data": {
    "values": [
      {"lat": 47.608013, "sunshine": 69},
      {"lat": 47.608013, "sunshine": 108},
      {"lat": 41.881832, "sunshine": 135},
      {"lat": 41.881832, "sunshine": 187}
    ]
  },
  "encoding": {
    "x": {"field": "lat", "type": "quantitative"},
    "y": {"field": "sunshine", "type": "quantitative"}
  }
}
```

This spec produces this chart:
visualization

### LINKED ISSUE #5263: Support boxplot along continuous scale
Currently, it seems that the orientation of boxes for boxplots is inferred via which axis is set to type `ordinal` (IIUC).

However, there are use cases for boxplots over irregularly sampled points in time, e.g. at Mars, we study ten-thousands of surface deposits with these distributions of areas, here, at 3 different points in time (time is called Ls = solar longitude for orbiting planetary objects):

ithaca_histograms

resulting in plots that describe the development of these distributions over a range of time, like this plot, where we use error bars of interquartile ranges, to emulate boxplots:

Figure 1 (1)

The current norm for boxplots is to do this:

```json
{
  "data": {"url": "https://vega.github.io/vega-datasets/data/population.json"},
  "mark": {"type": "boxplot", "extent": "min-max"},
  "encoding": {
    "x": {"type": "ordinal", "field": "age"},
    "y": {"type": "quantitative", "field": "people"}
  }
}
```
creating:

visualization (1)

but in that case, when there'd be data missing over the x-axis, that void would just be filled, as x-axis isn't a quantity (as discussed here: https://github.com/altair-viz/altair/issues/1496).

Jake's work-around in that issue was to impose a domain scale with given bins like so:

```python
x=alt.X('age:O', scale=alt.Scale(domain=np.arange(0, 95, 5).tolist())),
```

but my proposed feature is to simply switch the x-axis type to `quantitative`, so that the boxes are being placed at given data points for x.

However, trying to do that currently, like so:

```json
{
  "data": {"url": "https://vega.github.io/vega-datasets/data/population.json"},
  "mark": {"type": "boxplot", "extent": "min-max"},
  "encoding": {
    "x": {"type": "quantitative", "field": "age"},
    "y": {"type": "quantitative", "field": "people"}
  }
}
```

creates an odd looking plot where bins seem to be merged:

visualization

because `vega-lite` doesn't know from where to infer the direction of the boxes.

I guess my feature request would then lead to an `orientation` keyword, possibly with a default of being drawn parallel to the y-axis.

My expected behavior/spec would be (note the `orientation` option for `mark`:

```json
{
  "data": {"url": "https://vega.github.io/vega-datasets/data/population.json"},
  "mark": {"type": "boxplot", "extent": "min-max", "orientation": "y"},
  "encoding": {
    "x": {"type": "quantitative", "field": "age"},
    "y": {"type": "quantitative", "field": "people"}
  }
}
```
and a plot that could exclude missing data, treating the x-axis quantitatively, like here:

61178556-f4fd3d00-a5a3-11e9-976a-9f864403b65f-1

### PR DESCRIPTION
**Pull request by domoritz**
fixes #5263, fixes #4849

Also detect the orientation of bar marks from aggregation.

### CODE REVIEW

**Comment by domoritz** [comment]
We could set it only if we have a q x q boxplot...

File: src/compile/mark/init.ts [hunk 1]
```diff
@@ -98,10 +98,10 @@ function orient(mark: Mark, encoding: Encoding<string>, specifiedOrient: Orienta
 
   switch (mark) {
     case BAR:
-      if (isFieldDef(x) && isBinned(x.bin)) {
+      if (isFieldDef(x) && (isBinned(x.bin) || (isFieldDef(y) && y.aggregate && !x.aggregate))) {
```
Comment (chanwutk) [inline src/compile/mark/init.ts]: Should we add the case where there are `x`m `y` and `y2` but not `x2`?

Comment (domoritz) [reply src/compile/mark/init.ts]: Can you give me an example? I don't know how to answer your questions but we should cover all reasonable cases.

Comment (chanwutk) [reply src/compile/mark/init.ts]: For example, if there are `x`, `y`, and `y2` in the encoding channel, it should be `"vertical"`
```javascript
{
  mark: "bar",
  encoding: {
    x: {type: "quantitative", field: "field_for_x"},
    y: {type: "quantitative", field: "field_for_y"},
    y2: {field: "field_for_y2"}
  }
}
```

Comment (domoritz) [reply src/compile/mark/init.ts]: Unfortunately, binned data looks the same: https://vega.github.io/vega-lite/docs/bin.html#binned and the orientation should be different.

Comment (chanwutk) [reply src/compile/mark/init.ts]: I see.

**Pull request merged**